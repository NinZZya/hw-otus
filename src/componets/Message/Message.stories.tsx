import React from "react";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { Message, MessageProp } from "./Message";

export default {
  title: "Components/Message",
  component: Message,
} as Meta;

export const MessageWithTitle: React.FC<MessageProp> = () => (
  <Message title="Title" />
);

export const MessageWithText: React.FC<MessageProp> = () => (
  <Message text="Text" />
);

export const MessageWithTitleAndText: React.FC<MessageProp> = () => (
  <Message title="Title" text="Text" />
);
