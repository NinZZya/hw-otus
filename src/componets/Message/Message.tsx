import React from "react";
import styled from "@emotion/styled";

export interface MessageProp {
  title?: string;
  text?: string;
}

export const MessageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  aligin-items: center;
  padding: 1rem;
  margin: 1rem;
  min-width: 200px;
  min-height: 150px;
  background-color: #222;
  border-radius: 0.5rem;
  color: #fff;
`;

export const Message: React.FC<MessageProp> = (props) => {
  const { title, text } = props;
  return (
    <MessageWrapper>
      {title && <h2>{title}</h2>}
      {text && <p>{text}</p>}
    </MessageWrapper>
  );
};
