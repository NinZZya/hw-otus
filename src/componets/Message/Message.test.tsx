import React from "react";
import renderer from "react-test-renderer";
import { Message } from "./Message";

it("Message render coorectly", () => {
  const tree = renderer.create(<Message />).toJSON();

  expect(tree).toMatchSnapshot();
});
