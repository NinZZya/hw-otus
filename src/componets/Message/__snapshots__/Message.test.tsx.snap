// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render coorectly 1`] = `
<div
  className="css-101b6q5"
/>
`;
